diff -Naur a/xbmc/Application.cpp b/xbmc/Application.cpp
--- a/xbmc/Application.cpp	2013-03-05 10:57:46.000000000 -0500
+++ b/xbmc/Application.cpp	2013-03-05 11:04:02.353933167 -0500
@@ -795,7 +795,7 @@
   uint32_t sdlFlags = 0;
 
 #if (defined(HAS_SDL_OPENGL) || (HAS_GLES == 2)) && !defined(HAS_GLX)
-  sdlFlags |= SDL_INIT_VIDEO;
+  //sdlFlags |= SDL_INIT_VIDEO;
 #endif
 
 #if defined(HAS_SDL_JOYSTICK) && !defined(TARGET_WINDOWS)
diff -Naur a/xbmc/input/linux/LinuxInputDevices.cpp b/xbmc/input/linux/LinuxInputDevices.cpp
--- a/xbmc/input/linux/LinuxInputDevices.cpp	2013-02-26 06:14:43.000000000 -0500
+++ b/xbmc/input/linux/LinuxInputDevices.cpp	2013-03-05 10:57:46.000000000 -0500
@@ -1066,6 +1066,13 @@
   /* fill device info structure */
   GetInfo(fd);
 
+  if (m_deviceType==LI_DEVICE_NONE)
+  {
+    CLog::Log(LOGERROR, "CLinuxInputDevice %s: Unknown device type.", m_deviceName);
+    close(fd);
+    return false;
+  }
+
   if (m_deviceType & LI_DEVICE_KEYBOARD)
     SetupKeyboardAutoRepeat(fd);
 
diff -Naur a/xbmc/system.h b/xbmc/system.h
--- a/xbmc/system.h	2013-03-05 10:57:46.000000000 -0500
+++ b/xbmc/system.h	2013-03-05 11:04:31.841795497 -0500
@@ -167,15 +167,16 @@
 #ifdef HAVE_SDL
 #define HAS_SDL
 #ifndef HAS_SDL_OPENGL
-#define HAS_SDL_OPENGL
+//#define HAS_SDL_OPENGL
 #endif
 #ifndef HAVE_X11
-#define HAS_SDL_WIN_EVENTS
+//#define HAS_SDL_WIN_EVENTS
 #endif
 #else
 #ifndef HAVE_X11
 #define HAS_LINUX_EVENTS
 #endif
+#define HAS_LINUX_EVENTS
 #endif
 #define HAS_LINUX_NETWORK
 #define HAS_LIRC
