diff -Naur a/xbmc/Application.cpp b/xbmc/Application.cpp
--- a/xbmc/Application.cpp	2013-01-25 13:42:34.000000000 -0500
+++ b/xbmc/Application.cpp	2013-02-05 02:27:27.177039848 -0500
@@ -793,7 +793,7 @@
   uint32_t sdlFlags = 0;
 
 #if defined(HAS_SDL_OPENGL) || (HAS_GLES == 2)
-  sdlFlags |= SDL_INIT_VIDEO;
+  //sdlFlags |= SDL_INIT_VIDEO;
 #endif
 
 #if defined(HAS_SDL_JOYSTICK) && !defined(TARGET_WINDOWS)
diff -Naur a/xbmc/system.h b/xbmc/system.h
--- a/xbmc/system.h	2013-01-25 13:42:34.000000000 -0500
+++ b/xbmc/system.h	2013-02-05 00:54:35.103661691 -0500
@@ -151,38 +151,41 @@
  *****************/
 
 #if defined(TARGET_LINUX) || defined(TARGET_FREEBSD)
-#if defined(HAVE_LIBAVAHI_COMMON) && defined(HAVE_LIBAVAHI_CLIENT)
-#define HAS_ZEROCONF
-#define HAS_AVAHI
-#endif
-#define HAS_LCD
-#ifdef HAVE_DBUS
-#define HAS_DBUS
-#endif
-#define HAS_GL
-#ifdef HAVE_X11
-#define HAS_GLX
-#endif
-#ifdef HAVE_SDL
-#define HAS_SDL
-#ifndef HAS_SDL_OPENGL
-#define HAS_SDL_OPENGL
-#endif
-#define HAS_SDL_WIN_EVENTS
-#else
-#define HAS_LINUX_EVENTS
-#endif
-#define HAS_LINUX_NETWORK
-#define HAS_LIRC
-#ifdef HAVE_LIBPULSE
-#define HAS_PULSEAUDIO
-#endif
-#ifdef HAVE_LIBXRANDR
-#define HAS_XRANDR
-#endif
-#ifdef HAVE_ALSA
-#define HAS_ALSA
-#endif
+	#if defined(HAVE_LIBAVAHI_COMMON) && defined(HAVE_LIBAVAHI_CLIENT)
+		#define HAS_ZEROCONF
+		#define HAS_AVAHI
+	#endif
+	#define HAS_LCD
+	#ifdef HAVE_DBUS
+		#define HAS_DBUS
+	#endif
+	#define HAS_GL
+	#ifdef HAVE_X11
+		#define HAS_GLX
+	#endif
+	#ifdef HAVE_SDL
+		#define HAS_SDL
+		#ifndef HAS_SDL_OPENGL
+			#define HAS_SDL_OPENGL
+		#endif
+		#ifdef HAVE_X11
+			#define HAS_SDL_WIN_EVENTS
+		#endif
+	#endif
+	#ifndef HAS_SDL_WIN_EVENTS
+		#define HAS_LINUX_EVENTS
+	#endif
+	#define HAS_LINUX_NETWORK
+	#define HAS_LIRC
+	#ifdef HAVE_LIBPULSE
+		#define HAS_PULSEAUDIO
+	#endif
+	#ifdef HAVE_LIBXRANDR
+		#define HAS_XRANDR
+	#endif
+	#ifdef HAVE_ALSA
+		#define HAS_ALSA
+	#endif
 #endif
 
 #ifdef HAVE_LIBSSH
